src/qml/QFieldSettings.qml

```diff
@@ -240,6 +241,64 @@ Page {
                           }
                       }
                   }
+
+                  GridLayout {
+                      Layout.fillWidth: true
+                      Layout.leftMargin: 20
+                      Layout.rightMargin: 20
+                      Layout.topMargin: 5
+
+                      columns: 1
+                      columnSpacing: 0
+                      rowSpacing: 5
+
+                      Label {
+                          Layout.fillWidth: true
+                          text: qsTr( "QField user interface language:" )
+                          font: Theme.defaultFont
+
+                          wrapMode: Text.WordWrap
+                      }
+
+                      Label {
+                          id: languageTip
+                          visible: false
+
+                          Layout.fillWidth: true
+                          text: qsTr( "To apply the selected user interface language, QField needs to completely shutdown and restart." )
+                          font: Theme.tipFont
+                          color: Theme.warningColor
+
+                          wrapMode: Text.WordWrap
+                      }
+
+                      ComboBox {
+                          id: languageComboBox
+                          enabled: true
+                          Layout.fillWidth: true
+                          Layout.alignment: Qt.AlignVCenter
+
+                          property variant languages: {"":"System language","en":"English","bg":"български (Bulgarian)","zh":"中文 (Chinese)","fi":"Suomi (Finnish)","fr":"Français (French)","gl":"Galego (Galician)","de":"Deutsch (German)","hu":"Magyar (Hungarian)","ja":"日本 (Japanese)","pl":"Polskie (Polish)","pt":"Português (Portuguese)","pt_BR":"Português/Brasil (Portuguese/Brazil)","ru":"Pусский (Russian)","es":"Español (Spanish)","tr":"Türk (Turkish)"}
```

Comment:
Ah, can we have this list generated somehow and coming from somewhere else? This will be PIA to maintain. In the worst case scenarion, at least add it as a multiline JSON.

Reply:
Open to ideas on how this would be implemented here. I couldn't think of a clean way to do this.

Reply:
Q_INVOKABLE QMap<QString> AppInterface::supportedLanguages() seems to merely move the arbitrary list to C++, I don't think it's worth adding a json file to resources.qrc (let alone check if we could open this from QML javascript).

Not sure here :)

Reply:
I vote to defer this specific change to a follow up PR when we'll handle automatic generation of translations.qrc, it'd be an ideal time to also figure out a way to automate that QML list.

Reply:
I agree this can be done in follow-up PR. Please make it multi-line for now, so it's at least readable what we have in the list.